games/applications/hidden_shift.py

```diff
@@ -0,0 +1,147 @@
+# """Hidden Shift Benchmark Circuit"""
+
+# initialization
+import numpy as np
+import os
+import pytest
+from qiskit import IBMQ, Aer, QuantumCircuit, transpile, assemble
+from qiskit.visualization import plot_histogram
+# from applications import backends, run_qiskit_circuit
+
+#create qasm files
+QASM_DIR = os.path.join(os.path.dirname(os.path.abspath(__file__)), "qasm//hs/")
+
+#------------------------------------------------------------
+
+num_qubits = 6               # number of input qubits
+
+secret_string = "010110"     # the binary hidden shift string 
+
+#------------------------------------------------------------
+
+def the_shift(num_qubits, secret_string):
+    # the shifting operation for the given secret_string
+    the_shift = QuantumCircuit(num_qubits)
+    secret_string = secret_string[::-1]    # reverse secret_string to fit qiskit'secret_string qubit ordering
+    for i_qubit in range(num_qubits):
+        if secret_string[i_qubit] == '0':   # skips every secret_string qubit == 0
+            continue
+        else:
+            the_shift.x(i_qubit)    # applies x gate to secret_string qubit == 1, shifts the qubits
+    the_shift.name = 'Shift'
+    return the_shift
+
+# Display circuit
+the_shift(num_qubits, secret_string).draw()     # display the_shift
+
+#------------------------------------------------------------
+
+def g_oracle(num_qubits):
```

Comment:
Pass secret_string as argument
```suggestion
def g_oracle(secret_string, num_qubits):
```